docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,309 @@
+claim might be reused across multiple consumers over time.  Claims are abstract
+by nature (see the storage and networking forms of this diagram below) and
+describe what kind of resource is needed, but (generally) not the identity of
+the specific resource instance.  A `Manager` entity (a process, not part of the
+API) is responsible for satisfying the claim by "binding" the claim to a
+specific resource.
+
+You can sort of see this same pattern in the scheduling of `Pod`s onto `Node`s.
+A user creates a `Pod`, which is a claim against resources on a `Node`, and is
+bound by the scheduler.
+
+## Storage
+
+Storage in Kubernetes will be represented by `PersistentVolume`s.  A user can
+create a `PersistentVolumeClaim` for storage and use that claim from a `Pod`.
+A `VolumeManager` is responsible for binding `PersistentVolumeClaim`s to
+`PersistentVolume`s.  If the `PersistentVolumeClaim` can not be satisified by
+any existing `PersistentVolume`, the `VolumeManager` may provision a new one.
+
+```
+                       +------------------+
+        +--create----> | PersistentVolume |
+        |              +------------------+
+   +----+----+               ^
+   | VolMgr  |---bind------> :
+   +----+----+               v
+        |              +------------------------+
+        +---watch----> | PersistentVolumeClaim  | <--+
+                       +------------------------+    |
+                                 ^                   |
+                                 |         +---------+-- user-created
+                                 |         |
+                           +----------+    |
+                           |    Pod   | <--+
+                           +----------+
+```
+
+### PersistentVolumeClaim
+
+A `PersistentVolumeClaim` does not specify which specific PersistentVolume it
+wants or even which volume plugin should be used.  Instead it specifies
+characteristics of the storage it needs and those characteristics are satisfied
+as best is possible.  Characteristics include the minimum storage space
+required and the minimum access mode required (e.g. multi-writer).
+
+As the ecosystem of volume plugins grows, there will be an ever-increasing
+number of things that could be specified about volumes.  Rather than growing
+`PersistentVolumeClaim` to cover the union of every volume's parameters, users
+will be able to request coarse "classes" of volume (e.g. Gold, Silver, Bronze).
+How those classes map to real storage technologies is a policy set by the
+cluster administrators.
+
+### VolumeManagers
+
+The act of binding a `PersistentVolumeClaim` to a `PersistentVolume` is the
+responsibility of a process in the cluster.  A cluster may have any number of
+`VolumeManager`s running, each of which handles some set of claim classes.
+When a `PersistentVolumeClaim` is created that asks for "Gold" storage, the
+`VolumeManager` which understands the "Gold" class will make the binding.
```

Comment:
How soon do you expect the class stuff to be implemented? We need to make it consistent with multiple schedulers #11793, auto-scalers, etc. I was considering to use initializer classes for that. https://github.com/kubernetes/kubernetes/issues/3585#issuecomment-132855944

cc @erictune